packages/marshal/src/helpers/safe-promise.js

```diff
@@ -0,0 +1,61 @@
+// @ts-check
+
+/// <reference types="ses"/>
+
+import { isPromise } from '@endo/promise-kit';
+import { assertChecker } from './passStyle-helpers.js';
+
+/** @typedef {import('../types.js').Checker} Checker */
+
+const { details: X, quote: q } = assert;
+const { isFrozen, getPrototypeOf } = Object;
+const { ownKeys } = Reflect;
+
+/**
+ * Under Hardened JS a promise is "safe" if its `then` menthod can be called
+ * syncronously without giving the promise an opportunity for a
+ * reentrancy attack during that call.
+ *
+ * https://github.com/Agoric/agoric-sdk/issues/9
+ * raises the issue of testing that a specimen is a safe promise
+ * such that the test also does not give the specimen a
+ * reentrancy opportunity. That is well beyond the ambition here.
+ * TODO Though if we figure out a nice solution, it might be good to
+ * use it here as well.
+ *
+ * @param {unknown} pr The value to examine
+ * @param {Checker} [check]
+ * @returns {pr is Promise} Whether it is a safe promise
+ */
+const checkSafePromise = (pr, check = x => x) => {
+  let keys;
+  return (
+    check(isPromise(pr), X`${pr} - Must be a promise`) &&
+    check(isFrozen(pr), X`${pr} - Must be frozen`) &&
```

Comment:
As a matter of hygiene, I'd like the frozen status check to come first. And I think our normal pattern with assert/check messages is to put the variable parts last.
```suggestion
    check(isFrozen(pr), X`A safe promise must be frozen: ${pr}`) &&
    check(isPromise(pr), X`A safe promise must be a promise: ${pr}`) &&
```

Reply:
Agreed. Done.